googletest/src/gtest.cc

```diff
@@ -2979,13 +2979,16 @@ void ColoredPrintf(GTestColor color, const char* fmt, ...) {
   CONSOLE_SCREEN_BUFFER_INFO buffer_info;
   GetConsoleScreenBufferInfo(stdout_handle, &buffer_info);
   const WORD old_color_attrs = buffer_info.wAttributes;
-
+  // Let's reuse the BG
+  const WORD background_mask = BACKGROUND_BLUE | BACKGROUND_GREEN | BACKGROUND_RED | BACKGROUND_INTENSITY;   
+  const WORD existing_bg = old_color_attrs & background_mask;
+  
   // We need to flush the stream buffers into the console before each
   // SetConsoleTextAttribute call lest it affect the text that is already
   // printed but has not yet reached the console.
   fflush(stdout);
   SetConsoleTextAttribute(stdout_handle,
-                          GetColorAttribute(color) | FOREGROUND_INTENSITY);
+                          GetColorAttribute(color) | existing_bg | FOREGROUND_INTENSITY);
```

Comment:
Ok. I'll group them together. :)

That's a good question. So, if foreground == background, should we just invert the color bits (or just invert the FOREGROUND_INTENSITY bit)? That will hopefully work.